integration-tests/jdbc-repository/src/main/resources/application.properties

```diff
@@ -5,3 +5,14 @@ quarkus.jberet.repository.type=jdbc
 quarkus.jberet.repository.jdbc.datasource=batch
 
 quarkus.native.additional-build-args =-H:ReflectionConfigurationFiles=reflection-config.json
+
+quarkus.arc.selected-alternatives=io.quarkiverse.jberet.it.jdbc.JdbcRepositoryTest
+
+%test-prefix-suffix.quarkus.jberet.repository.jdbc.db-table-prefix=jb_prefix_
+%test-prefix-suffix.quarkus.jberet.repository.jdbc.db-table-suffix=_jb_suffix
+%test-prefix-suffix.job-instance-tablename=jb_prefix_JOB_INSTANCE_jb_suffix
+
+%test-ddl-filename.quarkus.jberet.repository.jdbc.ddl-file=ddl/custom-jberet-tables.ddl
+%test-ddl-filename.quarkus.jberet.repository.jdbc.sql-file=ddl/custom-jberet-sql.properties
+%test-ddl-filename.job-instance-tablename=jb_test_schema.JOB_INSTANCE
+%test-ddl-filename.quarkus.native.resources.includes=ddl/**
```

Comment:
Ah, in cases like this it is better if you use the `NativeImageResourceBuildItem` to register additional resources with the native image. Check an example here:

https://github.com/quarkiverse/quarkus-jberet/blob/87f8d07793a0290534d84909c558f9e91ef0a711/core/deployment/src/main/java/io/quarkiverse/jberet/deployment/JBeretProcessor.java#L183-L190

Reply:
I have never worked with these components.
What I am supposed to do? Just create a method in jdbc-repository test project, annotate with @BuildStep and Quarkus will do the magic?

Reply:
No, these annotations only work at build time and in `deployment` like extensions.

In this case, it is only a matter of updating that method. Just inject the configuration (adding a parameter with `JBeretConfig` is enough). Then you can check if there is any configuration setup with custom ddl and register the files referenced by the configuration with a `NativeImageResourceBuildItem`.

If you don't feel comfortable with that change, no worries, I can take it from here :) You did great!

Reply:
Custom resources for JDBC repositories are registered during native image production.
I'm still having troubles for native image tests because tests fail with strange errors similar to:
... property quarkus.jberet.repository.jdbc.sql-file was 'null' during build phase, now is 'dl/custom-jberet-tables.ddl'; build properties cannot be changed during runtime.

Do you have some advices?

Reply:
Yes. This is because these configurations are set to be built and runtime fixed. When you build the native image, the configuration properties are not available, so they are not recorded in the native image.

Then in the test, the configuration changes: 
https://github.com/quarkiverse/quarkus-jberet/blob/d8aa7b615b1506b44b057342d5242983546c8593/integration-tests/jdbc-repository/src/test/java/io/quarkiverse/jberet/it/jdbc/JdbcRepositoryTablesPrefixSuffixTest.java#L22-L29

Reply:
Ok. BUILD_AND_RUNTIME_FIXED in JBeretConfig is quite esplicative.
So, if I understand correctly, profiles and native images and incompatible? 
Do I have to disable @NativeImageTests because I need profiles for tests?

Reply:
You could keep them but don't run them with a native image (so remove the native image test for those cases)

On the side, you can add a native image test that doesn't require a profile (just set up the configuration in the `application.properties` file).